From: Ben Hutchings <ben@decadent.org.uk>
Subject: Revert "time: Avoid making adjustments if we haven't accumulated
 anything"
Date: Sun, 27 Jan 2013 18:42:52 +0000
Bug-Debian: http://bugs.debian.org/699112

This reverts commit 61b76840ddee647c0c223365378c3f394355b7d7
(upstream commit bf2ac312195155511a0f79325515cbb61929898a).

In 2.6.32 we still need to update xtime_cache regularly.  The
optimisation of update_wall_time() breaks that, causing some programs
to run slowly.

---
--- a/kernel/time/timekeeping.c
+++ b/kernel/time/timekeeping.c
@@ -807,10 +807,6 @@ void update_wall_time(void)
 #else
 	offset = timekeeper.cycle_interval;
 #endif
-	/* Check if there's really nothing to do */
-	if (offset < timekeeper.cycle_interval)
-		return;
-
 	timekeeper.xtime_nsec = (s64)xtime.tv_nsec << timekeeper.shift;
 
 	/* normally this loop will run just once, however in the
